### ISSUE TITLE
high cpu usage (up to 100%)

### ISSUE THREAD

**Issue by waateekaa** [wailsapp/wails#302]
**Description**

Just create a demo app (react version has the worst cpu usage)

**To Reproduce**
Steps to reproduce the behaviour:
1. wails init ... wait until build is done
2. start the built executable, the app and a WebkitWebProcess comes up
3. open top or similar tool
4. WebKitWebProcess takes 100% cpu or more (plus the demo about 15%) - see screenshot
5. Closing the app the WebkitWebProcess disappears as well and everything is normal as before.

I tested it on a different computer and noticed that the cpu usage is below 100% but always around 25...30% for any of the demos (angular, react, vue) which is still too much.
Tested with node 10.x as well, same behaviour.

**Expected behaviour**
I expect a app running in idle mode not to consume this amount of cpu. I expect less than 1% cpu usage in idle mode.

**Screenshots**
![wails-demo-cpu-usage](https://user-images.githubusercontent.com/38429404/70164641-d1a56c80-16c1-11ea-819a-d084a59accf0.jpg)

**System Details**
Ubuntu 18.04.3 LTS
| Name   | Value |
| ----- | ----- |
| Wails Version | v0.19.0 |
| Go Version    | go1.13.1 |
| Platform      | linux |
| Arch          | amd64 |
| GO111MODULE   | auto |
| GCC           | 7.4.0 |
| Npm           | 6.4.1 |
| Node          | v8.16.0 |

**Comment by bh90210** [comment]
react in specific has animation. the circling logo.
if you remove this it should be better.

 what is your total CPU? I could not tell from the screenshot

**Comment by waateekaa** [comment]
Hi, 
sorry it's german language. :-D be == Benutzer == user
so "%CPU(s):  2,2 be," means "%CPU(s):  2,2 user",

Yes the rotating logo takes a lot of cpu. Seems there's something wrong with the animation.
The values are from my laptop. Meanwhile I did tests with my desktop pc. The WebkitWebProcess does not consume so much cpu. So the webkit issue could  be related to the configuration or a library version. Anyway, the  demos consume always between 15 and 20% cpu. And this is too much in my opinion.

Please note also the virtual space of 82,941g.

Here are the top outputs (angular, vue2, react)
 
`**WAILS ANGULAR**

top - 21:16:18 up 6 min,  1 user,  load average: 0,50, 0,68, 0,36
Tasks: 244 gesamt,   1 laufend, 178 schlafend,   0 gestoppt,   0 Zombie
%CPU(s):  2,2 be,  2,7 sy,  0,0 ni, 95,1 un,  0,1 wa,  0,0 hi, 0,0 si, 0,0 st
KiB Spch : 32888244 gesamt, 28834256 frei,  1377356 belegt, 2676632 Puff/Cache
KiB Swap: 33497084 gesamt, 33497084 frei,        0 belegt. 30957360 verfü Spch

   PID USER      PR  NI    VIRT    RES    SHR S  %CPU %MEM ZEIT+ BEFEHL
  6598 user      20   0 82,941g  80436  64932 S  18,9  0,2 0:15.07 wails-angular
  2063 root      20   0  439884  76180  60476 S   2,0  0,2 0:11.93 Xorg
  2608 user      20   0  344672  25580  19852 S   1,7  0,1 0:04.71 panel-14-system
  4901 user      20   0  581544  41740  30292 S   0,7  0,1 0:02.67 xfce4-terminal

**WAILS VUE2**

top - 21:19:09 up 9 min,  1 user,  load average: 0,34, 0,54, 0,36
Tasks: 245 gesamt,   1 laufend, 179 schlafend,   0 gestoppt,   0 Zombie
%CPU(s):  2,3 be,  2,8 sy,  0,0 ni, 94,8 un,  0,0 wa,  0,0 hi, 0,1 si, 0,0 st
KiB Spch : 32888244 gesamt, 28527336 frei,  1378268 belegt, 2982640 Puff/Cache
KiB Swap: 33497084 gesamt, 33497084 frei,        0 belegt. 30959728 verfü Spch

   PID USER      PR  NI    VIRT    RES    SHR S  %CPU %MEM ZEIT+ BEFEHL
  6975 user      20   0 82,950g  86448  68636 S  18,9  0,3 0:02.89 wails-vue2
  2608 user      20   0  344672  25580  19852 S   1,7  0,1 0:07.51 panel-14-system
  2063 root      20   0  443120  79916  63828 S   1,0  0,2 0:18.49 Xorg
  5159 user      20   0  887336 210364 132424 S   0,7  0,6 0:11.76 chrome
  2369 user      20   0  201780  25080  19792 S   0,3  0,1 0:02.04 xfwm4
  2373 user      20   0  440500  48252  25748 S   0,3  0,1 0:01.34 xfce4-panel

**WAILS REACT**

top - 21:21:05 up 11 min,  1 user,  load average: 1,11, 0,71, 0,44
Tasks: 245 gesamt,   1 laufend, 179 schlafend,   0 gestoppt,   0 Zombie
%CPU(s):  7,7 be,  7,1 sy,  0,0 ni, 85,0 un,  0,2 wa,  0,0 hi, 0,1 si, 0,0 st
KiB Spch : 32888244 gesamt, 28124128 frei,  1419836 belegt, 3344280 Puff/Cache
KiB Swap: 33497084 gesamt, 33497084 frei,        0 belegt. 30920272 verfü Spch

   PID USER      PR  NI    VIRT    RES    SHR S  %CPU %MEM ZEIT+ BEFEHL
  7309 user      20   0 98,873g  85968  65928 S  26,8  0,3 0:03.65 wails-react
  2063 root      20   0  443124  77316  61228 S  13,6  0,2 0:23.53 Xorg
  7323 user      20   0 82,305g 135612  95632 S   7,9  0,4 0:01.05 WebKitWebProces
  4901 user      20   0  581792  41992  30360 S   3,0  0,1 0:06.14 xfce4-terminal

`

Thank you!

**Comment by leaanthony** [comment]
Hey there. Thanks for opening this. I'd be keen to hear other people's experience with CPU usage. I know there was a similar issue opened up for WebView, the windowing library we use but I can't find it.

**Comment by waateekaa** [comment]
Hi,
meanwhile I did some little tests. Maybe this can help to find the root cause. 
I linked against wails@v0.17.15-pre (Don't ask why exactly this version, it's just the one I still have on disk).
The CPU usages is less than 5% for the angular and the vue version and about 20% for the react version. But because of the animation it's not comparable to the others I think.
If I click the "hello" button the cpu usage goes up to something between 8 and 10% for a short time but then goes down to 2% or less in idle mode.

**Comment by waateekaa** [comment]
Hi,
I've tested multiple versions. It happens from v0.18.3 to v0.18.4.

**Comment by leaanthony** [comment]
Thanks for your persistence on this. Are you saying pre v0.18.3 it's fine?

**Comment by waateekaa** [comment]
Hi Lea,
pre 18.4 is fine. 
By example:
The angular app build with 18.3 has a cpu usage between 0 and 5% when idle.
The same app build with 18.4 the cpu usage is always between 18 and 25% in idle mode.
Blame it to the new signal handler????

**Comment by leaanthony** [comment]
I absolutely couldn't reproduce this. What might be useful is if you ran your app with tracing enabled. This should tell you where it's spending the most time.

**Comment by leaanthony** [comment]
Any updates @waateekaa ?

**Comment by waateekaa** [comment]
oh, I can see my response wasn't delivered.
I added trace logs. I will split it into two parts.
/cheers

Am Sa., 18. Jan. 2020 um 20:25 Uhr schrieb Lea Anthony <
notifications@github.com>:

**Comment by waateekaa** [comment]
Here are the trace file. I never did go trace analysis before. So I can't tell what's wrong. Hopefully you can.

I updated to the latest wails version, added 
trace.Start(os.Stderr)
defer trace.Stop()
to the main function and ran the react and angular demo app version. Please find the trace files attached.

[wails-angular-trace-v1.01.out.zip](https://github.com/wailsapp/wails/files/4086496/wails-angular-trace-v1.01.out.zip)
[wails-react-trace-v1.01.out.zip](https://github.com/wailsapp/wails/files/4086497/wails-react-trace-v1.01.out.zip)

**Comment by leaanthony** [comment]
Hey, sorry for not getting onto this yet. Will look this weekend

**Comment by leaanthony** [comment]
Have not forgotten about this! I'm guessing the situation is still the same and the traces are still valid?

**Comment by waateekaa** [comment]
Hi,
I just checked out wails version 1.0.2.
And here are the good news. Seems the issue is solved. I did tests on my laptop and on my pc 
for the angular, the react and the vue2 app demo. All of them do not consume as much cpu as before. I got values less than 3% cpu usage for the angular and vue version which is very good and about 15% for the react version (blame it to the animation).
I would treat this issue as resolved.
Great job!
